.github/workflows/ci.yml

```diff
@@ -146,12 +199,18 @@ jobs:
       - uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
-      - run: yarn install --frozen-lockfile --ignore-engines
-      - run: yarn add pg-native --ignore-engines
-        if: matrix.native
-      - name: Unit Tests
-        run: yarn test-unit
-        if: ${{ !matrix.minify-aliases }}
+          cache: yarn
+      - uses: actions/download-artifact@v2
+        with:
+          name: install-build-artifact-node-${{ matrix.node-version }}
+      - name: Extract artifact
+        run: tar -xf install-build-node-${{ matrix.node-version }}.tar
+      - name: Install pg-native (Node 12)
+        run: yarn add pg-native --ignore-engines
+        if: matrix.native && matrix.node-version == 12
+      - name: Install pg-native (Node 16)
+        run: npm install --no-save --no-audit pg-native
```

Comment:
we should document this npm-based hack to be safe

Reply:
I've included `# This step uses npm instead of yarn to minimize the time needed. See #14171`. Is that enough?

Reply:
Yup!